python/ray/util/client/common.py

```diff
@@ -285,11 +286,25 @@ class ClientActorHandle(ClientStub):
 
     def __init__(self, actor_ref: ClientActorRef):
         self.actor_ref = actor_ref
+        self._dir: Optional[List[str]] = None
 
     def __del__(self) -> None:
         if ray.is_connected():
             ray.call_release(self.actor_ref.id)
 
+    def __dir__(self) -> List[str]:
+        if self._dir:
+            return self._dir
+        if ray.is_connected():
+
+            @client_mode_wrap
```

Comment:
Are we guaranteed to be in client mode here?
If so, I think it would be clearer to directly use a cpu=0 ray task.

Reply:
Makes sense!

Reply:
Calling a remote function and `ray.get` inside of `__dir__` still feels like an unreasonable thing to do. Is there another way to make sure the handle has the appropriate information?